tmt/steps/provision/__init__.py

```diff
@@ -59,6 +60,254 @@ class CheckRsyncOutcome(enum.Enum):
+        # cast: since the default is None, mypy cannot infere the full type,
+        # and reports `package_manager` parameter to be `object`.
+        default=cast(Optional[GuestPackageManager], None),
+        serialize=lambda package_manager: package_manager.value if package_manager else None,
+        unserialize=lambda raw_value: GuestPackageManager(raw_value) if raw_value else None)
+
+    has_selinux: Optional[bool] = None
+
+    os_release_content: Dict[str, str] = field(default_factory=dict)
+    lsb_release_content: Dict[str, str] = field(default_factory=dict)
+
+    # TODO nothing but a fancy helper, to check for some special errors that
+    # may appear this soon in provisioning. But, would it make sense to put
+    # this detection into the `GuestSsh.execute()` method?
+    def _execute(
+            self,
+            guest: 'Guest',
+            command: Command) -> Optional[tmt.utils.CommandOutput]:
+        """
+        Run a command on the given guest.
+
+        On top of the basic :py:meth:`Guest.execute`, this helper is able to
+        detect a common issue with guest access. Facts are the first info tmt
+        fetches from the guest, and would raise the error as soon as possible.
+
+        :returns: command output if the command quit with a zero exit code,
+            ``None`` otherwise.
+        :raises tmt.units.GeneralError: when logging into the guest fails
+            because of a username mismatch.
+        """
+
+        try:
+            return guest.execute(command, silent=True)
+
+        except tmt.utils.RunError as exc:
+            if exc.stdout and 'Please login as the user' in exc.stdout:
+                raise tmt.utils.GeneralError(f'Login to the guest failed.\n{exc.stdout}') from exc
+
+        return None
+
+    def _fetch_keyval_file(self, guest: 'Guest', filepath: Path) -> Dict[str, str]:
+        """
+        Load key/value pairs from a file on the given guest.
+
+        Converts file with ``key=value`` pairs into a mapping. Some values might
+        be wrapped with quotes.
+
+        .. code:: shell
+
+           $ cat /etc/os-release
+           NAME="Ubuntu"
+           VERSION="20.04.5 LTS (Focal Fossa)"
+           ID=ubuntu
+           ID_LIKE=debian
+           ...
+
+        :returns: mapping with key/value pairs loaded from ``filepath``, or an
+            empty mapping if it was impossible to load the content.
+        """
+
+        content: Dict[str, str] = {}
+
+        output = self._execute(guest, Command('cat', str(filepath)))
+
+        if not output or not output.stdout:
+            return content
+
+        for line in output.stdout.splitlines(keepends=False):
+            line = line.strip()
+            if not line:
+                continue
+
+            key, value = line.split('=', 1)
+
+            if value.startswith('"'):
+                value = value[1:]
+            if value.endswith('"'):
+                value = value[:-1]
```

Comment:
Why not use `value.strip('"')`?

Reply:
`strip('"')` would remove all leading and trailing quotes, but what do I know, `FOO="Debian "TM""` might be valid key/value pair.

Reply:
Implementation changed,  I need to resolve this in order to be able to merge.